tket/test/src/Circuit/test_Circ.cpp

```diff
@@ -3117,6 +3117,185 @@ SCENARIO("check edge type in rewire function") {
     REQUIRE_THROWS_AS(c.rewire(v, ins, types), CircuitInvalidity);
   }
 }
+void check_conditional_circuit(Circuit& c) {
+  // Confirm that the DAG is constructed appropriately by checking
+  // in and out edges of all vertices
+  std::vector<Vertex> vertices = c.all_vertices();
+  // First check Quantum and Classical input and output vertices
+  // have expected number of out edges in interesting cases
+  REQUIRE(c.n_out_edges_of_type(vertices[8], EdgeType::Classical) == 1);
```

Comment:
This test is pretty difficult to review; have you checked by visual inspection as well?

Reply:
Yeah apologies - I visually checked and it's right, and i wanted to way to preserve that.